### PR TITLE
feat: Implemented SOL signMessage function

### PR DESCRIPTION
**Pull request by Dorac**
----

*Checklist:*

- Unit tests have been verified and passed successfully using the following input data:
`To avoid digital dognappers, sign below to authenticate with CryptoCorgis`

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-19858 Implement signMessage method](https://app.clickup.com/t/25577573/CW-19858)

File: packages/coin-sol/src/config/types.ts [hunk 1]
```diff
@@ -167,7 +167,7 @@ export type SignInMessage = {
   resources?: string[];
 };
 
-type Mandatory<T> = T extends SignInMessage
+type Mandatory<T> = T extends SignInMessage | Uint8Array
```
Comment (gundomlegend) [inline packages/coin-sol/src/config/types.ts:170]: 感覺可以直接讓上層喂 string 進去比較方便，轉 Uint8Array 是有什麼特殊原因嗎？

Comment (Dorac) [reply packages/coin-sol/src/config/types.ts:170]: 官方文件的介面建議hex或Uint8Array，可以改hex string，前端再幫忙轉

Comment (gundomlegend) [reply packages/coin-sol/src/config/types.ts:170]: 了解～前端收到的應該都是 hexString

**Review by gundomlegend — DISMISSED** [review]

**Review by gundomlegend — APPROVED** [review]

**Review by hankliu418 — APPROVED** [review]

**Pull request merged**